Bug: asoctl panic during import

**Describe the bug**

Importing azure resources can crash with a panic:

```
panic: reflect: call of reflect.Value.SetString on zero Value

goroutine 3028 [running]:
reflect.flag.mustBeAssignableSlow(0x0?)
        /home/bevan/.gvm/gos/go1.20.4/src/reflect/value.go:255 +0x116
reflect.flag.mustBeAssignable(...)
        /home/bevan/.gvm/gos/go1.20.4/src/reflect/value.go:249
reflect.Value.SetString({0x0?, 0x0?, 0xc0007eb790?}, {0xc000514334, 0x7})
        /home/bevan/.gvm/gos/go1.20.4/src/reflect/value.go:2437 +0x4f
github.com/Azure/azure-service-operator/v2/cmd/asoctl/internal/importing.(*importableARMResource).SetName(0xc000b321cb?, {0x7f8f90bef480, 0xc00091e5a0}, {0xc000514334, 0x7}, {{0xc0007240c0, 0x11}, {0x4178c1e, 0xe}, {0xc000b72000, ...}, ...})
        /home/bevan/github/azure-service-operator/v2/cmd/asoctl/internal/importing/importable_arm_resource.go:484 +0x1ae
github.com/Azure/azure-service-operator/v2/cmd/asoctl/internal/importing.(*importableARMResource).createImportableObjectFromID(0xc00065a570, 0xc000ae8cc0, 0xc000b0c240)
        /home/bevan/github/azure-service-operator/v2/cmd/asoctl/internal/importing/importable_arm_resource.go:374 +0x23c
github.com/Azure/azure-service-operator/v2/cmd/asoctl/internal/importing.(*importableARMResource).importResource(0xc00065a570, {0x55a8ef8, 0xc00011e2d0}, 0xc000b0c240)
        /home/bevan/github/azure-service-operator/v2/cmd/asoctl/internal/importing/importable_arm_resource.go:140 +0x7c
github.com/Azure/azure-service-operator/v2/cmd/asoctl/internal/importing.(*importableARMResource).Import(0xc00065a570, {0x55a8ef8, 0xc00011e2d0}, 0x3?)
        /home/bevan/github/azure-service-operator/v2/cmd/asoctl/internal/importing/importable_arm_resource.go:94 +0x79
github.com/Azure/azure-service-operator/v2/cmd/asoctl/internal/importing.(*ResourceImporter).ImportResource(0xc0001d0150, {0x55a8ef8, 0xc00011e2d0}, {0x55a90b8?, 0xc00065a570?})
        /home/bevan/github/azure-service-operator/v2/cmd/asoctl/internal/importing/resource_importer.go:188 +0x54e
github.com/Azure/azure-service-operator/v2/cmd/asoctl/internal/importing.(*ResourceImporter).Import.func1()
        /home/bevan/github/azure-service-operator/v2/cmd/asoctl/internal/importing/resource_importer.go:136 +0x48
golang.org/x/sync/errgroup.(*Group).Go.func1()
        /home/bevan/.gvm/pkgsets/go1.20.4/global/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
        /home/bevan/.gvm/pkgsets/go1.20.4/global/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
```
**To Reproduce**

``` bash
asoctl import azure-resource <resource-group>
```

The crash occurs if the resource group being imported contains any resource that has a hardcoded name. For example, `Microsoft.Storage/storageAccounts/tableServices` must always be called `default` so we can't set the name.

**Expected behavior**

Import should work without crashing.

--- a/v2/cmd/asoctl/internal/importing/importable_arm_resource.go
+++ b/v2/cmd/asoctl/internal/importing/importable_arm_resource.go
@@ -484,11 +484,16 @@ func (i *importableARMResource) SetName(
 
 	importable.SetName(n)
 
-	// AzureName needs to be exactly as specified in the ARM URL
-	// Need to use reflection to set it
+	// AzureName needs to be exactly as specified in the ARM URL.
+	// Use reflection to set it as we don't have convenient access.
+	// Not all resources have the AzureName property - some resources
+	// have hard coded names that ASO handles directly
+	// (e.g. Microsoft.Storage/storageAccounts/tableServices always has the name 'default')
 	specField := reflect.ValueOf(importable.GetSpec()).Elem()
 	azureNameField := specField.FieldByName("AzureName")
+	if azureNameField.IsValid() {
 	azureNameField.SetString(name)
+	}
 }
 
 func (i *importableARMResource) SetOwner(

--- a/v2/cmd/asoctl/internal/importing/importable_arm_resource.go
+++ b/v2/cmd/asoctl/internal/importing/importable_arm_resource.go
@@ -352,7 +352,7 @@ func (*importableARMResource) classifyError(err error) (string, bool) {
 func (i *importableARMResource) createImportableObjectFromID(
 	owner *genruntime.ResourceReference,
 	armID *arm.ResourceID,
-) (genruntime.ImportableARMResource, error) {
+) (resource genruntime.ImportableARMResource, failure error) {
 	gvk, err := i.groupVersionKindFromID(armID)
 	if err != nil {
 		return nil, errors.Wrap(err, "unable to determine GVK of resource")
@@ -368,6 +368,13 @@ func (i *importableARMResource) createImportableObjectFromID(
 		return nil, errors.Errorf(
 			"unable to create blank resource, expected %s to identify an importable ARM object", armID)
 	}
+
+	defer func() {
+		if err := recover(); err != nil {
+			resource = nil
+			failure = errors.Errorf("Internal error: %s", err)
+		}
+	}()
 
 	if owner != nil {
 		i.SetOwner(importable, *owner)

--- a/v2/cmd/asoctl/internal/importing/importable_arm_resource.go
+++ b/v2/cmd/asoctl/internal/importing/importable_arm_resource.go
@@ -492,7 +492,7 @@ func (i *importableARMResource) SetName(
 	specField := reflect.ValueOf(importable.GetSpec()).Elem()
 	azureNameField := specField.FieldByName("AzureName")
 	if azureNameField.IsValid() {
-	azureNameField.SetString(name)
+		azureNameField.SetString(name)
 	}
 }
 

--- a/v2/cmd/asoctl/internal/importing/importable_arm_resource.go
+++ b/v2/cmd/asoctl/internal/importing/importable_arm_resource.go
@@ -352,14 +352,21 @@ func (*importableARMResource) classifyError(err error) (string, bool) {
 func (i *importableARMResource) createImportableObjectFromID(
 	owner *genruntime.ResourceReference,
 	armID *arm.ResourceID,
-) (resource genruntime.ImportableARMResource, failure error) {
-	gvk, err := i.groupVersionKindFromID(armID)
-	if err != nil {
+) (resource genruntime.ImportableARMResource, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			resource = nil
+			err = errors.Errorf("creating importable object for %s: %s", armID.String(), r)
+		}
+	}()
+
+	gvk, gvkErr := i.groupVersionKindFromID(armID)
+	if gvkErr != nil {
 		return nil, errors.Wrap(err, "unable to determine GVK of resource")
 	}
 
-	obj, err := i.createBlankObjectFromGVK(gvk)
-	if err != nil {
+	obj, objErr := i.createBlankObjectFromGVK(gvk)
+	if objErr != nil {
 		return nil, errors.Wrap(err, "unable to create blank resource")
 	}
 
@@ -368,13 +375,6 @@ func (i *importableARMResource) createImportableObjectFromID(
 		return nil, errors.Errorf(
 			"unable to create blank resource, expected %s to identify an importable ARM object", armID)
 	}
-
-	defer func() {
-		if err := recover(); err != nil {
-			resource = nil
-			failure = errors.Errorf("Internal error: %s", err)
-		}
-	}()
 
 	if owner != nil {
 		i.SetOwner(importable, *owner)

--- a/v2/cmd/asoctl/internal/importing/resource_importer_report.go
+++ b/v2/cmd/asoctl/internal/importing/resource_importer_report.go
@@ -6,6 +6,7 @@
 package importing
 
 import (
+	"errors"
 	"sync"
 
 	"github.com/go-logr/logr"
@@ -124,11 +125,28 @@ func (k *resourceImportReportKey) lessThan(other resourceImportReportKey) bool {
 }
 
 func (k *resourceImportReportKey) WriteToLog(log logr.Logger, count int) {
-	log.Info(
-		"Summary",
-		"Group", k.group,
-		"Kind", k.kind,
-		"Status", k.status,
-		"Count", count,
-		"Reason", k.reason)
+	switch k.status {
+	case Imported:
+		log.Info(
+			"Successful imports",
+			"Group", k.group,
+			"Kind", k.kind,
+
+			"Count", count,
+			"Reason", k.reason)
+	case Skipped:
+		log.V(1).Info(
+			"Skipped imports",
+			"Group", k.group,
+			"Kind", k.kind,
+			"Count", count,
+			"Reason", k.reason)
+	case Failed:
+		log.Error(
+			errors.New(k.reason),
+			"Failed imports",
+			"Group", k.group,
+			"Kind", k.kind,
+			"Count", count)
+	}
 }

--- a/v2/cmd/asoctl/internal/importing/resource_importer_report.go
+++ b/v2/cmd/asoctl/internal/importing/resource_importer_report.go
@@ -131,9 +131,7 @@ func (k *resourceImportReportKey) WriteToLog(log logr.Logger, count int) {
 			"Successful imports",
 			"Group", k.group,
 			"Kind", k.kind,
-
-			"Count", count,
-			"Reason", k.reason)
+			"Count", count)
 	case Skipped:
 		log.V(1).Info(
 			"Skipped imports",

